### PR TITLE
[#64] -1 for non-selected indicies;

### PR DESCRIPTION
**Pull request by kirilknysh**
Fixes https://github.com/21-23/_qd-ui/issues/64

### CODE REVIEW

File: app/game/middleware/server-pipe.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ function formatStateMessage(message) {
             displayName,
         },
         session: {
-            currentRoundIndex: puzzleIndex,
+            currentRoundIndex: Number.isFinite(puzzleIndex) ? puzzleIndex : -1,
```
Comment (DaQuirm) [inline app/game/middleware/server-pipe.js]: That means NO IE SUPPORT?!

File: app/common/components/game-progress/game-progress.js [hunk 2]
```diff
@@ -35,6 +35,10 @@ function getMarkText(index, currentRoundIndex) {
 }
 
 function getProgressFactor(currentRoundIndex, totalRounds) {
+    if (currentRoundIndex < 0) {
```
Comment (DaQuirm) [inline app/common/components/game-progress/game-progress.js:38]: Not the scope of this PR, but this should be called `puzzleIndex`, right?

File: app/game/middleware/server-pipe.js (hunk 1 above)
Comment (kirilknysh) [reply app/game/middleware/server-pipe.js]: 🤔

**Review by DaQuirm — APPROVED** [review]

File: app/common/components/game-progress/game-progress.js (hunk 2 above)
Comment (kirilknysh) [reply app/common/components/game-progress/game-progress.js:38]: it's mapped to it. Just to be in sync with `currentRoundIndex` and `selectedRoundIndex`

**Pull request merged**